server/src/main/java/org/opentosca/toscana/plugins/cloudformation/visitor/TransformModelNodeVisitor.java

```diff
@@ -193,108 +236,41 @@ public void visit(WebApplication node) {
-
-            // Add file to config
-            cfnModule.getCFNInit(serverName)
-                .getOrAddConfig(CONFIG_SETS, config)
-                .putFile(cfnFile);
-        }
-
-        //Add artifact
-        if (operation.getArtifact().isPresent()) {
-            String artifact = operation.getArtifact().get().getFilePath();
-            String cfnSource = getFileURL(cfnModule.getBucketName(), artifact);
-
-            logger.debug("Marking '{}' as file to be uploaded.", artifact);
-            cfnModule.putFileToBeUploaded(artifact);
-            if (!cfnModule.getAuthenticationSet().contains(serverName)) {
-                logger.debug("Marking '{}' as instance in need of authentication.", serverName);
-            } else {
-                logger.debug("'{}' already marked as instance in need of authentication. " +
-                    "Skipping authentication marking.", serverName);
-            }
-            cfnModule.putAuthentication(serverName);
-
-            CFNFile cfnFile = new CFNFile(ABSOLUTE_FILE_PATH + artifact)
-                .setSource(cfnSource)
-                .setMode(MODE_500) //TODO Check what mode is needed (read? + execute?)
-                .setOwner(OWNER_GROUP_ROOT) //TODO Check what Owner is needed
-                .setGroup(OWNER_GROUP_ROOT); //TODO Check what Group is needed
-
-            CFNCommand cfnCommand = new CFNCommand(artifact,
-                ABSOLUTE_FILE_PATH + artifact) //file is the full path, so need for "./"
-                .setCwd(ABSOLUTE_FILE_PATH + new File(artifact).getParent());
-            // add inputs to environment
-            for (OperationVariable input : operation.getInputs()) {
-                String value = input.getValue().orElseThrow(
-                    () -> new IllegalArgumentException("Input value of " + input.getKey() + " expected to not be " +
-                        "null")
-                );
-                if (cfnModule.checkFn(value)) {
-                    cfnCommand.addEnv(input.getKey(), cfnModule.getFn(value));
-                } else {
-                    cfnCommand.addEnv(input.getKey(), value);
-                }
-            }
-            // Add file to config and execution command
-            cfnModule.getCFNInit(serverName)
-                .getOrAddConfig(CONFIG_SETS, config)
-                .putFile(cfnFile)
-                .putCommand(cfnCommand);
+    @Override
+    public void visit(Nodejs node) {
+        try {
+            Compute computeHost = getCompute(node);
+            String computeHostName = toAlphanumerical(computeHost.getEntityName());
+            
+            //handle configure
+            operationHandler.handleConfigure(node, computeHostName);
+            //handle start
+            operationHandler.handleStart(node, computeHostName);
+            //add NodeJs create script
+            operationHandler.addCreate(FILEPATH_NODEJS_CREATE, computeHostName);            
+            
+            //Open port 3000
```

Comment:
Please consider reading the port from the node, instead of hardcoding port 3000. 
This could be achieved like this (Taken from the DockerfileBuildingVisitor):
```java
node.getCapabilities().forEach(e -> {
 try {
   if (e instanceof EndpointCapability) {
     if (((EndpointCapability) e).getPort().isPresent()) {
       ports.add(((EndpointCapability) e).getPort().get().port);
     }
   }
 } catch (Exception ex) {
   logger.warn("Failed reading Port from node {}", node.getEntityName(), ex);
 }
});
```